sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/AppendBlobClient.java

```diff
@@ -108,6 +111,28 @@ public BlobOutputStream getBlobOutputStream() {
         return getBlobOutputStream(null);
     }
 
+    /**
+     * Creates and opens an output stream to write data to the append blob.
+     * <p>
+     *
+     * @return A {@link BlobOutputStream} object used to write data to the blob.
+     * @param overwrite Whether an existing blob should be deleted and recreated, should data exist on the blob.
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public BlobOutputStream getBlobOutputStream(boolean overwrite) {
+        AppendBlobRequestConditions requestConditions = null;
+        if (!overwrite) {
+            if (exists()) {
```

Comment:
Do we want to throw here or put the OutputStream into append mode?

Reply:
So it looks like in [.NET](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs#L1984) when overwrite is false and blob exists, they continue to append. We should probably follow the same logic.